libs/service/api_service.cpp

```diff
@@ -432,6 +442,206 @@ ebpf_verify_and_load_program(
+        if (signature_settings.dwIndex != index) {
+            cleanup_win_verify_trust();
+
+            win_trust_data.dwStateAction = WTD_STATEACTION_VERIFY;
+            win_trust_data.pSignatureSettings->dwIndex = index;
+
+            DWORD error = WinVerifyTrust(nullptr, &generic_action_code, &win_trust_data);
+            if (error != ERROR_SUCCESS) {
+                EBPF_LOG_WIN32_API_FAILURE(EBPF_TRACELOG_KEYWORD_API, WinVerifyTrust);
+                throw std::runtime_error("WinVerifyTrust failed");
+            }
+        }
+
+        CRYPT_PROVIDER_DATA* provider_data = WTHelperProvDataFromStateData(win_trust_data.hWVTStateData);
+        CRYPT_PROVIDER_SGNR* provider_signer = WTHelperGetProvSignerFromChain(provider_data, 0, FALSE, 0);
+        CRYPT_PROVIDER_CERT* cert = WTHelperGetProvCertFromChain(provider_signer, 0);
+
+        if (cert == nullptr) {
+            EBPF_LOG_WIN32_API_FAILURE(EBPF_TRACELOG_KEYWORD_API, WTHelperGetProvCertFromChain);
+            throw std::runtime_error("WTHelperGetProvCertFromChain failed");
+        }
+        return cert;
+    }
+
+  private:
+    void
+    cleanup_win_verify_trust()
+    {
+        if (win_trust_data.hWVTStateData) {
+            win_trust_data.dwStateAction = WTD_STATEACTION_CLOSE;
+            WinVerifyTrust(nullptr, &generic_action_code, &win_trust_data);
+            win_trust_data.hWVTStateData = nullptr;
+        }
+    }
+
+    GUID generic_action_code = WINTRUST_ACTION_GENERIC_VERIFY_V2;
+    WINTRUST_DATA win_trust_data = {};
+    WINTRUST_FILE_INFO file_info = {};
+    WINTRUST_SIGNATURE_SETTINGS signature_settings = {};
+};
+
+static std::set<std::string>
+_ebpf_extract_eku(_In_ CRYPT_PROVIDER_CERT* cert)
+{
+    std::set<std::string> eku_set;
+    DWORD cb_usage = 0;
+    if (!CertGetEnhancedKeyUsage(cert->pCert, 0, nullptr, &cb_usage)) {
+        return eku_set;
+    }
+    std::vector<uint8_t> usage(cb_usage);
+
+    if (!CertGetEnhancedKeyUsage(cert->pCert, 0, reinterpret_cast<PCERT_ENHKEY_USAGE>(usage.data()), &cb_usage)) {
+        return eku_set;
+    }
+    auto pusage = reinterpret_cast<PCERT_ENHKEY_USAGE>(usage.data());
+    for (size_t index = 0; index < pusage->cUsageIdentifier; index++) {
+        eku_set.insert(pusage->rgpszUsageIdentifier[index]);
+    }
+    return eku_set;
+}
+
+static std::string
+_ebpf_extract_issuer(_In_ CRYPT_PROVIDER_CERT* cert)
+{
+    DWORD name_cb =
+        CertGetNameStringA(cert->pCert, CERT_KEY_PROV_INFO_PROP_ID, CERT_NAME_ISSUER_FLAG, nullptr, nullptr, 0);
+
+    std::vector<char> issuer(name_cb);
+    if (CertGetNameStringA(
+            cert->pCert, CERT_KEY_PROV_INFO_PROP_ID, CERT_NAME_ISSUER_FLAG, nullptr, issuer.data(), name_cb) == 0) {
+        return std::string();
+    }
+    return std::string(issuer.data());
+}
+
+_Must_inspect_result_ ebpf_result_t
+ebpf_verify_sys_file_signature(
+    _In_z_ const wchar_t* file_name,
+    _In_z_ const char* issuer_name,
+    _In_ size_t eku_count,
```

Comment:
```suggestion
    size_t eku_count,
```
use `_In_` with reference parameters, not value parameters.